linera-views/src/common.rs

```diff
@@ -378,11 +387,14 @@ pub trait AdminKeyValueStore: Sized {
     async fn list_all(config: &Self::Config) -> Result<Vec<String>, Self::Error>;
 
     /// Deletes all the existing namespaces.
-    async fn delete_all(config: &Self::Config) -> Result<(), Self::Error> {
-        for namespace in Self::list_all(config).await? {
-            Self::delete(config, &namespace).await?;
+    fn delete_all(config: &Self::Config) -> impl Future<Output = Result<(), Self::Error>> {
```

Comment:
Is that really absolutely needed?
The `async fn`  syntax is familiar and working well.

Reply:
See the comment [here](https://github.com/linera-io/linera-protocol/pull/1843#discussion_r1546426560) — it's because `trait-variant` gets confused with default implementations of `async fn`s.

Reply:
(I would also prefer to keep it as `async fn` if we could! I think we should undo this change as soon as `trait-variant` supports default-implemented `async fn`s.)